chaoslib/litmus/disk_fill_by_litmus.yml

```diff
@@ -0,0 +1,161 @@
+    fill: "{{ fill_percent |float / 100 }}"
+
+- name: Fill percent into storage
+  set_fact:
+    to_fill: "{{ fill|float * limit_KB|float }}"
+
+- name: Get the Container ID
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.status.containerStatuses[?(@.name=="{{ app_container }}")].containerID}' | awk -F'//' '{print $2}'
+  args:
+    executable: /bin/bash
+  register: containerID
+
+- name: Identify the application node
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} --no-headers -o custom-columns=:spec.nodeName
+  args:
+    executable: /bin/bash
+  register: app_node
+
+- name: Create the DaemonSet for Disk-Fill
+  shell: >
+    kubectl apply -f /experiments/generic/disk_fill/disk_fill_ds.yaml -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: ds_status
+  
+- name: Confirm that the disk-fill daemonSet is running on all nodes
+  shell: >
+    kubectl get pods -l app=disk-fill --no-headers -o custom-columns=:status.phase -n {{ namespace }} | sort | uniq
+  args:
+    executable: /bin/bash
+  register: result
+  until: "result.stdout == 'Running'"
+  delay: 1
+  retries: 60
+
+- name: Record the disk-fill pod on app node
+  shell: >
+    kubectl get pods -l app=disk-fill -o wide -n {{ namespace }} | grep {{ app_node.stdout }} | awk '{print $1}'
+  args:
+    executable: /bin/bash
+  register: disk_fill_pod
+
+- name: Exec into the pod to get storage size
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ namespace }} -- bash -c "cd /diskfill && cd {{ containerID.stdout }} && du -h |tail -n1 | awk '{print $1}' | grep -Eo '+|[0-9]+'"
+  args:
+    executable: /bin/bash
+  register: pod_current_size
+
+- name: Exec into the pod to get storage unit
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ namespace }} -- bash -c "cd /diskfill && cd {{ containerID.stdout }} && du -h |tail -n1 | awk '{print $1}' | grep -Eo '[[:alpha:]]+'"
+  args:
+    executable: /bin/bash
+  register: pod_current_storage_unit
+
+- name: Fetch the value of Used Storage Unit in KB
+  set_fact:
+    used_storage_KB: "{{ lookup('vars', pod_current_storage_unit.stdout) }}"
+
+- name: Total size of Used Storage in KB
+  set_fact:
+    used_storage: "{{ pod_current_size.stdout|int * used_storage_KB|int }}"
+
+- name: Subtract to get the file allocation size
+  set_fact:
+    fill_storage: "{{ to_fill|int - used_storage|int }}"
+
+- name: Block size for file creation
+  set_fact:
+    bs: 4K
+
+- name: Count for file creation 
+  set_fact:
+    fill_count: "{{ fill_storage|int / 4 }}"
+
+- name: Change count for file creation to integer
+  set_fact:
+    fill_count_int: "{{ fill_count|int }}"
+
+- name: File creation
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ namespace }} -- bash -c "cd /diskfill/{{ containerID.stdout }} && dd if=/dev/urandom of=/diskfill/{{ containerID.stdout }}/diskfill bs={{ bs }} count={{ fill_count_int }}"
```

Comment:
Interested to know about the choice of dd (which is not bad IMO) over fallocate. Also, clearly - this is a sepcific action , i..e., identifying to-fill value and file creation. Can go into a separate util, at least the blocksize and dd part!